### PR TITLE
Add gmsh to docker image

### PR DESCRIPTION
**Pull request by JDBetteridge**
~DO NOT MERGE (yet)~ 

I think this is ready to go

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -70,6 +70,27 @@ jobs:
         run: |
           cd ..
           rm -rf build
+  docker_test:
+    name: "Build Docker environment container"
```
Comment (connorjward) [inline .github/workflows/build.yml:74]: This is the same comment as for `docker_env`. What's the difference? (and maybe make it clear in a comment or something)

Comment (JDBetteridge) [reply .github/workflows/build.yml:74]: This is temporary, I just need a way of generating a docker image that isn't one that CI or users use

File: demos/linear-wave-equation/linear_wave_equation.py.rst [hunk 2]
```diff
@@ -107,7 +107,7 @@ options at this point, we may either `lump` the mass, which reduces
 the inversion to a pointwise division::
 
       if lump_mass:
-          p += assemble(dt * inner(nabla_grad(v), nabla_grad(phi))*dx) / assemble(v*dx)
+          p += interpolate(assemble(dt * inner(nabla_grad(v), nabla_grad(phi))*dx) / assemble(v*dx), V)
```
Comment (JDBetteridge) [inline demos/linear-wave-equation/linear_wave_equation.py.rst:110]: @connorjward can you update me on what assign is doing now. Do we still need this fix?

Comment (connorjward) [reply demos/linear-wave-equation/linear_wave_equation.py.rst:110]: I haven't made any recent changes to `assign`. I got quite confused when the adjoint got involved so stopped working on it.

Comment (JDBetteridge) [reply demos/linear-wave-equation/linear_wave_equation.py.rst:110]: In which case can we accept this as a temporary fix? If so, I'll create an issue documenting the issue and tidy up this PR

**Comment by JDBetteridge** [comment]
NB: This will need to go through the wash twice before the tests that require gmsh will run

File: .github/workflows/build.yml (hunk 1 above)
Comment (JDBetteridge) [reply .github/workflows/build.yml]: Gone

**Review by dham — APPROVED** [review]

**Pull request merged**